src/components/DatePicker/YearPicker/YearPicker.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from "react";
+import moment from "moment";
+import times from "lodash/times";
+import { CSSTransition } from "react-transition-group";
+import classnames from "classnames";
+import Button from "../../Button/Button";
+import { YEAR_FORMAT } from "../constants";
+import DateNavigationItemComponent from "../DateNavigationItem/DateNavigationItem";
+import { Moment, Direction } from "../types";
+import styles from "./YearPicker.module.scss";
+
+const transitionOptions = {
+  classNames: "slide-down",
+  timeout: { enter: 400, exit: 400 }
+};
+
+const PAGE_SIZE = 18;
+const BUFFER_FROM_CURRENT_YEAR = 4;
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const NOOP = () => {};
+
+interface YearPickerProps {
+  selectedDate?: Moment;
+  isYearBlocked?: (year: number) => boolean;
+  changeCurrentDate: (date: Moment) => void;
+  "data-testid"?: string;
+}
+
+const YearPicker = (props: YearPickerProps) => {
+  const calcNewYearsPage = (firstYearInPage: number) => {
+    return times(PAGE_SIZE, n => firstYearInPage + n);
+  };
+
+  const { selectedDate, isYearBlocked, changeCurrentDate, "data-testid": dateTestId } = props;
+  const selectedYear = selectedDate ? selectedDate.format(YEAR_FORMAT) : moment().format(YEAR_FORMAT);
+
+  const [yearsToDisplay, setYearsToDisplay] = useState(
+    calcNewYearsPage(parseInt(selectedYear) - BUFFER_FROM_CURRENT_YEAR)
+  );
+
+  const onYearNavigationClick = (direction: Direction) => {
+    const firstYearInPage = yearsToDisplay[0];
+    let newYearsArray: number[] = [];
+    if (direction === Direction.prev) {
+      newYearsArray = calcNewYearsPage(firstYearInPage - PAGE_SIZE);
+    } else if (direction === Direction.next) {
+      newYearsArray = calcNewYearsPage(firstYearInPage + PAGE_SIZE);
+    }
+    setYearsToDisplay(newYearsArray);
+  };
+
+  const onYearSelect = (year: number) => {
+    changeCurrentDate(moment().year(year));
+  };
+
+  const renderYears = () => {
+    const listItems = yearsToDisplay.map(currYear => {
+      const shouldBlockYear = isYearBlocked && isYearBlocked(currYear);
+      const onClick = !shouldBlockYear ? () => onYearSelect(currYear) : NOOP;
+      const kind = parseInt(selectedYear, 10) === currYear ? Button?.kinds?.PRIMARY : Button?.kinds?.TERTIARY;
+
+      return (
+        <Button key={currYear} kind={kind} onClick={onClick} disabled={shouldBlockYear} marginLeft marginRight>
+          {currYear.toString()}
+        </Button>
+      );
+    });
+
+    return listItems;
+  };
+
+  return (
+    <div data-testid={`${dateTestId}-year-picker`} className={styles.monthYearPickerContainer}>
```

Comment:
I would just say that in general we trying to use semantic html tags when it's possible (for accessibility) so if you see here a tag you can replace with a tag which represent better the element purpose, please replace it :)